src/main.cpp

```diff
@@ -1,259 +1,295 @@
-//             break;
-//           }
-//           case '1':
-//           {
-//             // serial.println('1');
-//             if(RTCConnected){
-//               // serial.print("Date and Time is: ");
-//               // serial.println(getTimeString());
-//             }
-//             else
-//             {
-//               // serial.println("RTC Module not initialized.");
-//             }
-//             break;
-//           }
-//           case '2':
-//           {
-//             if(SDConnected)
-//             {
-//               // serial.println('2');
-//               printFiles();
-//             } 
-//             else
-//             {
-//               // serial.println("SD Card not initialized.");
-//             }
-//             break;
-//           }
-//           case '3':
-//           {
-//             // serial.println('3');
-//             // serial.println("Set Sensor Read Interval Time: ");
-//             printMultiString(intervalString);
-//             state = 3;
-//             break;
-//           }
-//           case '4':
-//           {
-//             // serial.println('4');
-//             // serial.println("Set the Date and Time using this format... ");
-//             break;
-//           }
-//           default:
-//           {
-//             // serial.println("Unkown command, press h to view all commands. ");
-//             break;
-//           }
-//         }
-//         break;
-//       }
-//       // State 3, Set Interval Time
-//       case 3:
-//       {
-//         switch(input)
-//         {
-//           case '1':
-//           {
-//             // serial.println("Interval Set to: 5 Seconds");
-//             distanceReadInterval = 5000UL;
-//             state = 0;
-//             break;
-//           }
-//           case '2':
-//           {
-//             // serial.println("Interval Set to: 30 Seconds");
-//             distanceReadInterval = 30000UL;
-//             state = 0;
-//             break;
-//           }
-//           case '3':
-//           {
-//             // serial.println("Interval Set to: 2 Minutes");
-//             distanceReadInterval = 120000UL;
-//             state = 0;
-//             break;
-//           }
-//           case '4':
-//           {
-//             // serial.println("Interval Set to: 5 Seconds");
-//             distanceReadInterval = 3600000UL;
-//             state = 0;
-//             break;
-//           }
-//           default:
-//           {
-//             // serial.println("Invalid Input. Here Are Your Options: ");
-//             printMultiString(intervalString);
-//             break;
-//           }
-//         }
-//         break;
-//       }
-//       default:
-//       {
-//         break;
-//       }
-//     }
-//   }
-// }
+void cmdHandler()
+{
+	if (Serial.available() > 0)
+	{
+		char input = Serial.read();
+		switch (state)
+		{
+			case 0:
+			{
+				switch (input)
+				{
+					case 'h':
```

Comment:
This is probably fine since this has already been made, but I wonder if we could have used a terminal UI library like [ncrurses?](https://tldp.org/HOWTO/NCURSES-Programming-HOWTO/intro.html#WHATIS)